### ISSUE TITLE
AptPlot stream step does not give y-axis labels for NRC Databank data

### ISSUE THREAD

**Issue by AndrewIrelandNRC** [NRC-Research/SNAP-issues#95]
If you create an engineering template, add a job stream with an AptPlot step, and try to plot a NRC databank file, you get a plot with no y-axis labels. X-axis labels are fine.

**Comment by WDunsford** [comment]
Can you give me the databank file you're working with?  When I tried it locally the Y axis labels showed up, so it may be a file issue.

**Comment by AndrewIrelandNRC** [comment]
Sounds like probable user error, but I've tried a few databank files without luck. I sent some files via Box for you to check. In this case, I am using a RBHT databank file.

**Comment by NRCgg** [comment]
> Sounds like probable user error, but I've tried a few databank files without luck. I sent some files via Box for you to check. In this case, I am using a RBHT databank file.

@NRC-Research/snap & @AndrewIrelandNRC 👉  Just a reminder, the SNAP developers have access to RESGC, so no need to distribute information to them via BOX.  Just coordinate on where to find it in the RESGC filesystem!  For smaller files or code, you may also consider creating a private GitHub repository and giving them access.

**Comment by AndrewIrelandNRC** [comment]
My problem seems to stem from how AptPlot batch processing works for databank files (and probably other file types). The commands follow this sequence:

**Open the databank file > use CALC to save a channel as a _variable_ (loses the units) > plot the _variable_**

When done this way, the units are scrubbed. What I'm looking for is a way to retain the channel units, so I don't have to set the axis label manually in a batch file or job stream. I've been trying to follow the scripting help files, but I haven't had any luck. (Maybe the "NREAD" command would be helpful, but I'm not understanding what that command actually _does_ or what objects it creates.)

**Comment by WDunsford** [comment]
OK sorry, I got my wires crossed on this one. This isn't so much of a bug as a fundamental flaw in how the batch commands are executed in AptPlot. I went down a rabbit hole trying to figure out if there's a quick way to retrieve that data, but it would require a significant overhaul of how the batch commands operate. 

Ultimately Andrew, you're correct in that the engineering unit label is lost by the "<variable> = N0_(chan-100)" command. The data structure that holds the floating point values doesn't have a field for holding either the X or Y engineering units, so by the time the "plotvar" command is executed, there's no source for engineering units. Further, any update to initialize unit strings would have to be applied to all AptPlot plug-ins independently.

At this point the best way to set the Y Axis label in the AptPlot job step is to explicitly set the values in the Axis attribute group for the plot. The budget for AptPlot modifications won't support the level of effort required.  I'm closing this issue, but will keep it in mind for GRAVE Plot, which is intended to replace AptPlot as the packaged plotting utility for SNAP.
